034-default-to-restricted-pod-security-profile.md

```diff
@@ -0,0 +1,157 @@
+* _Baseline_ which prevents some common security issues, but is minimally restrictive
+* _Restricted_ which heavily restricts the application permissions and follows the best hardening practices
+
+The detailed description of the profiles including of the list of allowed and not-allowed configurations for each profile is available in [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/pod-security-standards/).
+
+Together with the profiles it also introduces Pod Security Admission plugin which can enforce the security standards.
+From Kubernetes 1.23, these features moved to Beta and are enabled by default.
+By default it does not enforce any of the profiles.
+Enforcing a specific profile can be configured either in the Kubernetes APi server configuration or using namespace labels.
+Once enabled, it will not allow creation of Pods which do not have the security context configured as described by the enforced profile.
+You can learn more about the Pod Security Admission in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/pod-security-admission/)
+
+## Current situation
+
+Currently, Strimzi does by default only minimal configuration of the security context.
+When running on Kubernetes and using persistent storage, we configure the group which should be used for the storage.
+That is the only automatic configuration Strimzi does.
+
+Users can use the `.template` fields in the Strimzi custom resources to configure their own security context.
+That allows them to customize it to match their own requirements and policies.
+On OpenShift, the security context is automatically injected into the Pods when they are created based on the OpenShift SCC policies.
+This typically involves dropping some capabilities, using some random unprivileged user ID etc.
+In some cases, our users might also use similar systems to inject the security context using admission controllers and similar tools.
+
+The _baseline_ profile basically means that you just use the default settings without requesting any additional privileges or capabilities.
+So without any of the additional tooling mentioned above, Strimzi operators and the pods created by them today match the _baseline_ profile.
+But if you would try to run Strimzi in a cluster where the _restricted_ profile is enforced, it would not work out of the box.
+
+Apart from the Kaniko builder used for the Kafka Connect build, all our components are able to run under the _restricted_ profile.
+But the operator doesn't configure the appropriate security context and the admission plugin will reject them.
+Users have to _manually_ add the matching security context configuration into the `.template` sections of the custom resources and into the operator deployments to allow the pods to be created under the _restricted_ profile.
+To match the _restricted_ profile, users would need to add the following security context:
+
+```yaml
```

Comment:
should we show `template:pod:` on this example?

Reply:
It would be container template and that has always a bit different name => so I think it is better without it.